### ISSUE TITLE
Add codeowners tooling to this repository

### ISSUE THREAD

**Issue by mx-psi** [open-telemetry/opentelemetry-collector#11562]
We have some tooling on the opentelemetry-collector-contrib that allows us to automate some actions related to codeowners:
- [ping-codeowners-prs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/ping-codeowners-prs.yml)
- [ping-codeowners-on-new-issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/ping-codeowners-on-new-issue.yml)
- [ping-codeowners-issues.yml](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/ping-codeowners-issues.yml)
- [add-codeowners-to-pr](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/add-codeowners-to-pr.yml)

We can add the tooling one by one. Some of these may need additional tooling (more issue templates?):

- [ ] Ping code owners on PRs for a component
- [ ] Ping code owners on new issues
- [ ] Ping code owners on existing issues
- [ ] Add code owners as reviewers in a PR

**Comment by mowies** [comment]
i'd like to work on this :)

**Comment by mowies** [comment]
@mx-psi so after a quick review of the 4 links above, I noticed that probably only the issue related things will really work on this repo. That's because the [codeowners file](https://github.com/open-telemetry/opentelemetry-collector/blob/main/.github/CODEOWNERS) is not set up as granular here compared to the contrib repo so the PR related workflows, that would ping codeowners for specific folders would just generate a ton of notifications for the collector maintainers.

**Comment by mx-psi** [comment]
@mowies That's fair. There will be a long transition period in which a lot of things do not have a codeowner. I think we should ignore the `*` line and only trigger automation if we are on a more specific line.

I am working on a PR that should be ready on Wednesday adding more specific codeowners, maybe after that is ready it's a good time to see how this would work in practice

**Comment by mowies** [comment]
The automations only look for specific labels as components anyways, so they already ignore the `*` case.

**Comment by mowies** [comment]
For some of the automations to work, I would also need to adjust the existing issue templates and probably mostly exchange them for the ones from the contrib repo. Should I add that to this PR or rather create a separate one?

**Comment by mx-psi** [comment]
> The automations only look for specific labels as components anyways, so they already ignore the * case.

Ah, goes to show how little I know about this :sweat_smile: I don't understand the notifications issue you mention above. Could you explain with an example where we would get a lot of notifications?

> Should I add that to this PR or rather create a separate one?

Let's create a separate one for this

**Comment by mowies** [comment]
I thought at first, that the `*` in the codeowners file of this repo would be respected by the automations and since that's (almost) the only entry in the file, that would trigger as a fallback for all components that are not specifically listed, but that's not actually the case after digging a little deeper. Instead, if the specific component is not found in the codeowners file, nobody will get notified. So that's good I guess in terms of spam 😅

**Comment by mx-psi** [comment]
Makes sense! Then yeah, the tooling will not be super useful until we have a more fleshed out list of codeowners, but it will be good as a start.

**Comment by mowies** [comment]
Yeah I think it's still good to have them in place. As soon as the issue templates are updated, we at least get labels for components, which is already something 😄 
And since no notification spam will occur, I can also go ahead and add the PR related automations as well I think.

**Comment by mowies** [comment]
some more ideas around this that i will extend over the course of implementing this:
- i think a few tools and scripts could be moved to a central repo like https://github.com/open-telemetry/opentelemetry-go-build-tools so that they can be reused and don't need to be copy/pasted all over the place (i'm copying lots of things from contrib into core right now. for example pipelines, githubgen tool)
- ...

**Comment by mowies** [comment]
first PR for this is ready for review: https://github.com/open-telemetry/opentelemetry-collector/pull/11756

**Comment by mowies** [comment]
@dmathieu could you please list a few tools that you'd like to see moved?
I'm currently creating an issue in otel-collector-contrib to move githubgen and I guess I could also create a parent issue to move more tools.

**Comment by dmathieu** [comment]
Right now, we're using request codeowners review:
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/.github/workflows/request_codeowners_review.yml
Pinging codeowners on issues is also something we're interested in though.
